packages/components/src/tree-grid/index.js

```diff
@@ -82,8 +87,44 @@ function TreeGrid( { children, ...props }, ref ) {
 				);
 			}
 
-			// Focus is either at the left or right edge of the grid. Do nothing.
+			// Focus is either at the left or right edge of the grid.
 			if ( nextIndex === currentColumnIndex ) {
+				if ( keyCode === LEFT ) {
+					// Left:
+					// If a row is focused, and it is expanded, collapses the current row.
+					if ( activeRow?.ariaExpanded === 'true' ) {
+						onCollapseRow( activeRow?.dataset?.block );
+						event.preventDefault();
+						return;
+					}
+					// If a row is focused, and it is collapsed, moves to the parent row (if there is one).
+					const level = Math.max(
+						parseInt( activeRow?.ariaLevel ?? 1, 10 ) - 1,
+						1
+					);
+					const rows = Array.from(
+						treeGridElement.querySelectorAll( '[role="row"]' )
+					);
+					let parentRow = activeRow;
+					const currentRowIndex = rows.indexOf( activeRow );
+					for ( let i = currentRowIndex; i >= 0; i-- ) {
+						if ( parseInt( rows[ i ].ariaLevel, 10 ) === level ) {
+							parentRow = rows[ i ];
+							break;
+						}
+					}
+					getRowFocusables( parentRow )?.[ 0 ]?.focus();
+				} else {
+					// Right:
+					// If a row is focused, and it is collapsed, expands the current row.
+					if ( activeRow?.ariaExpanded === 'false' ) {
+						onExpandRow( activeRow?.dataset?.block );
+						event.preventDefault();
+						return;
+					}
+					// If a row is focused, and it is expanded, focuses the first cell in the row.
+					getRowFocusables( activeRow )?.[ 0 ]?.focus();
+				}
```

Comment:
I love the `left` behaviour!! It works so effortlessly and exactly as I'd expect! ✨ 

I've got a suggestion for the `right` behaviour.
Nothing happens if you hit `right` again after expanding a branch.
This felt a bit awkward to me, and I've realized that's because I'm used to how the VSCode file tree works.

Hitting `right` again after expanding would move the selection to the first child.
If the first child is a branch, `right` would either expand it, or move to its first child.
Once the first child is a leaf, hitting `right` wouldn't do anything anymore.
(Hope I've been clear 😅 )

This is imho super handy to traverse down the tree, in a specular way to the upward-traversal of `left`.

Reply:
Right arrow doing nothing in the rightmost column is from treegrid behavior. I'd certainly be open to adding more functionality if folks agree with the behavior.

https://www.w3.org/TR/wai-aria-practices/examples/treegrid/treegrid-1.html

<img width="760" alt="Screen Shot 2021-06-11 at 8 42 18 AM" src="https://user-images.githubusercontent.com/1270189/121712783-f3d73200-ca90-11eb-8fc7-39f8cf85e87a.png">

Reply:
I see, thanks for the explanation!
I'd certainly agree with the behaviour I've described earlier. 😄

Reply:
Looks like that's how a tree view works:
https://www.w3.org/TR/wai-aria-practices/#keyboard-interaction-22

So it might be determined by how https://github.com/WordPress/gutenberg/issues/32294 is progressed. I'm not sure how that can be moved forwards, but probably worth coming to a decision soon.

Reply:
Thanks, I revisited this and I caught some behavior I misimplemented. I should be keeping focus in the rightmost cell when expanded instead of looping to the first focus item. This is updated in https://github.com/WordPress/gutenberg/pull/32117/commits/7065ab6e560363e551b9852d4de4f4117835c8cc 

We can see this in the navigation list view:

Before:


https://user-images.githubusercontent.com/1270189/121932756-16b84f00-ccfa-11eb-9ecc-a3a01d3816fa.mp4


After:

https://user-images.githubusercontent.com/1270189/121932775-2041b700-ccfa-11eb-8ac7-1cb8f8c55219.mp4